python/lsst/faro/utils/matcher.py

```diff
@@ -220,7 +220,7 @@ def ellipticity(i_xx, i_xy, i_yy):
     return e, e1, e2
 
 
-def makeMatchedPhotom(data: Dict[str, List[CalibratedCatalog]]):
+def makeMatchedPhotom(data: Dict[str, List[CalibratedCatalog]], logger=None):
```

Comment:
There is a bit of a philosophical logging issue in this entire file. It's probably not something for you to solve but given that almost all of these logger messages (in this file) are at DEBUG level should the caller be in charge of whether log output is possible at all or should the person running the task code? I think I'd prefer it if a log message was always issued if I asked for DEBUG output so the philosophical question then becomes whether the logger should be a mandatory parameter (the logger is the same as the task code) or a package log (`lsst.utils.logging.getLogger(__name__)`) which is always set to a predictable value in this file and is then an `lsst.faro` logger explicitly. We could use that logger if the caller doesn't set one but then you have a problem with unpredictability of the logger name. I don't think this is something that should be sorted out on this ticket but I think @jeffcarlin and @bechtol should think about it.

Reply:
I've made a [ticket](https://jira.lsstcorp.org/browse/DM-34150) for this question so that we don't lose the thread.